src/coreComponents/constitutive/fluid/BlackOilFluidBase.hpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file BlackOilFluidBase.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_BLACKOILFLUIDBASE_HPP_
+#define GEOSX_CONSTITUTIVE_FLUID_BLACKOILFLUIDBASE_HPP_
+
+#include "constitutive/fluid/MultiFluidBase.hpp"
+#include "functions/TableFunction.hpp"
+
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+class BlackOilFluidBase : public MultiFluidBase
+{
+public:
+
+  using exec_policy = parallelDevicePolicy<>;
+
+  struct PhaseType
+  {
+    static constexpr integer OIL            = 0;
+    static constexpr integer GAS            = 1;
+    static constexpr integer WATER          = 2;
+    static constexpr integer MAX_NUM_PHASES = 3;
+  };
```

Comment:
We don't want a strongly typed enum because these constants are actually mainly used as indicies into arrays. On the other hand, we (almost?) never switch on them. But I do agree with the rest. I originally proposed `static constexpr integer` approach, but later I realized `enum` is a little better (especially the part about not requiring a definition). I've done a similar transformation in `RelativePermeabilityBase`, in case you want to keep it similar.